packages/effects-core/src/plugins/sprite/sprite-item.ts

```
@@ -152,9 +152,24 @@ export class SpriteComponent extends BaseRenderComponent {
   }
 
   override onDestroy (): void {
+    const textures = this.getTextures();
+
     if (this.item && this.item.composition) {
-      this.item.composition.destroyTextures(this.getTextures());
+      this.item.composition.destroyTextures(textures);
     }
+
+    textures.forEach(texture => {
+      const source = texture.source;
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)